src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -0,0 +1,233 @@
+
+  return new_process.UnicodeProcessRunner(centipede_path)
+
+
+def _get_reproducer_path(line):
+  """Get the reproducer path, if any."""
+  crash_match = _CRASH_REGEX.match(line)
+  if not crash_match:
+    return None
+
+  return crash_match.group(1)
+
+
+class Engine(engine.Engine):
+  """Centipede engine implementation."""
+
+  @property
+  def name(self):
+    return 'centipede'
+
+  def prepare(self, corpus_dir, target_path, build_dir):  # pylint: disable=unused-argument
+    """Prepare for a fuzzing session, by generating options. Returns a
+    FuzzOptions object.
+
+    Args:
+      corpus_dir: The main corpus directory.
+      target_path: Path to the target.
+      build_dir: Path to the build directory.
+
+    Returns:
+      A FuzzOptions object.
+    """
+    os.chmod(target_path, 0o775)
+    arguments = []
+    dict_path = dictionary_manager.get_default_dictionary_path(target_path)
+    if os.path.exists(dict_path):
+      arguments.append(f'--dictionary={dict_path}')
+
+    # Is it OK to create workdir with this function?
+    # workdir saves centipede-readable corpus&feature files, and crashes
+    workdir = self._create_temp_dir('workdir')
+    # Will the workdir always exist?
+    arguments.append(f'--workdir={workdir}')
+
+    # Will the corpus directory always exist?
+    Path(corpus_dir).mkdir(exist_ok=True)
+    arguments.append(f'--corpus_dir={corpus_dir}')
+
+    # Download sanitized binaries.
+    sanitized_binaries = glob.glob(
+        os.path.join(build_dir, '*_*', os.path.basename(target_path)))
+    arguments.append(f'--extra_binaries={",".join(sanitized_binaries)}')
+
+    return engine.FuzzOptions(corpus_dir, arguments, {})
+
+  def fuzz(self, target_path, options, reproducers_dir, max_time):  # pylint: disable=unused-argument
+    """Run a fuzz session.
+
+    Args:
+      target_path: Path to the target.
+      options: The FuzzOptions object returned by prepare().
+      reproducers_dir: The directory to put reproducers in when crashes
+          are found.
+      max_time: Maximum allowed time for the fuzzing to run.
+
+   Returns:
+      A FuzzResult object.
+    """
+    runner = _get_runner()
+    arguments = _DEFAULT_ARGUMENTS[:]
+    arguments.extend(options.arguments)
+    arguments.extend([
+        f'--binary={target_path}',
+    ])
+
+    fuzz_result = runner.run_and_wait(
+        additional_args=arguments, timeout=max_time + _CLEAN_EXIT_SECS)
+    log_lines = fuzz_result.output.splitlines()
+    fuzz_logs = '\n'.join(log_lines)
+
+    crashes = []
+    # Stats report is not available in Centipede yet
+    stats = None
+    for line in log_lines:
+      reproducer_path = _get_reproducer_path(line)
+      if reproducer_path:
+
+        crashes.append(
+            engine.Crash(reproducer_path, fuzz_logs, [],
+                         int(fuzz_result.time_executed)))
+        continue
+
+      #stats = _get_stats(line)
+
+    #if stats is None:
+    #  stats = {}
```

Comment:
Delete debug code please?

Reply:
I was hoping to use them as a placed holder, and uncomment them when `Centipede` supports stats report (see the comment above : ))
But I think you are right, it is better to delete them now.